packages/core/src/odata-common/request-builder/request-builder-base.ts

```diff
@@ -119,6 +119,17 @@ export abstract class MethodRequestBuilder<
     return this;
   }
 
+  /**
+   * Add custom request configs to the request.
+   *
+   * @param requestConfigs - Key-value pairs denoting additional custom request configs to be set in the request.
+   * @returns The request builder itself, to facilitate method chaining.
+   */
+  addCustomRequestConfigs(requestConfigs: Record<string, string>): this {
```

Comment:
To add to my other comment on "configs":

I have no strong opinion here, but we might want to expose the full word: `addCustomRequestConfiguration`.
Although, this is quite long... but consider that we also expose `addCustomQueryParameters` (and not params)